lib/src/main/java/com/orange/ods/compose/component/banner/OdsBanner.kt

```diff
@@ -0,0 +1,163 @@
+import com.orange.ods.compose.component.button.OdsTextButtonStyle
+import com.orange.ods.compose.component.utilities.BasicPreviewParameterProvider
+import com.orange.ods.compose.component.utilities.Preview
+import com.orange.ods.compose.component.utilities.UiModePreviews
+import com.orange.ods.compose.theme.OdsTheme
+import com.orange.ods.theme.OdsColors
+
+/**
+ * <a href="https://system.design.orange.com/0c1af118d/p/19a040-banners/b/497b77" class="external" target="_blank">ODS banners</a>.
+ *
+ *
+ * @param message text displayed in the banner.
+ * @param modifier modifiers for the Banner layout.
+ * @param image image display in the banner.
+ * @param imageContentDescription Optional image content description.
+ * @param buttonText Optional text of the second button in the banner. If not present, button will not be shown. If present, [onButton2Click] need to be  handle.
+ * @param button1Text principal button in the banner, it displays an [OdsTextButton] with the given [button1Text] as an action of the banner.
+ * @param actionOnNewLine whether or not action should be put on the separate line.
+ * @param divider add the line at the end of the banner.
+ * @param onButtonClick Optional handler for the button click.
+ * @param onButton1Click executed on action button2 click.
+ */
+@Composable
+@OdsComponentApi
+fun OdsBanner(
+    modifier: Modifier = Modifier,
+    message: String,
+    image: Painter? = null,
+    imageContentDescription: String? = null,
+    actionOnNewLine: Boolean = false,
+    divider: Boolean = true,
+    button1Text: String,
+    buttonText: String? = null,
+    onButtonClick: (() -> Unit)? = null,
+    onButton1Click: () -> Unit = {}
+) {
+    Column(
+        modifier = modifier
+            .fillMaxWidth()
+    ) {
+        Row(
+            verticalAlignment = Alignment.CenterVertically,
+            modifier = Modifier.fillMaxWidth()
+                .padding(top = dimensionResource(id = R.dimen.spacing_m),
+                    bottom = dimensionResource(id = R.dimen.spacing_s)),
+            horizontalArrangement = Arrangement.SpaceBetween,
+        ) {
+            if (!actionOnNewLine) {
+                image?.let {
+                    Image(
+                        painter = image,
+                        contentDescription = imageContentDescription,
+                        contentScale = ContentScale.Crop,// crop the image if it's not a square
+                        modifier = Modifier
+                            .padding(start = dimensionResource(id = R.dimen.spacing_m))
+                            .size(50.dp)
+                            .clip(CircleShape)
+                    )
+                }
+            }
+            Text(
+                modifier = Modifier
+                    .padding(
+                        start = dimensionResource(id = R.dimen.spacing_m),
+                        end = dimensionResource(id = R.dimen.spacing_s)
```

Comment:
If there is no image, the text should be left aligned but this is not the case. You probably should replace this start padding by an end padding applied to the image. Like that, the padding will be displayed only if there is an image.

```kotlin
                Image(
                    painter = image,
                    contentDescription = imageContentDescription,
                    contentScale = ContentScale.Crop,// crop the image if it's not a square
                    modifier = Modifier
                        .padding(
                            end = dimensionResource(id = R.dimen.spacing_m)
                        )
                        .size(50.dp)
                        .clip(CircleShape)
                )
```